python/ray/util/client/worker.py

```diff
@@ -320,6 +334,25 @@ def _call_schedule_for_task(
                 logger.exception("Failed to deserialize {}".format(
                     ticket.error))
                 raise
+        self.total_num_tasks_scheduled += 1
+        self.total_outbound_message_size_bytes += task.ByteSize()
+        if self.total_num_tasks_scheduled > TASK_WARNING_THRESHOLD and \
+                log_once("client_communication_overhead_warning"):
+            logger.warning(
+                f"More than {TASK_WARNING_THRESHOLD} remote tasks have been "
+                "scheduled. This can be slow on Ray Client due to "
+                "communication overhead. If you're running many fine-grained "
+                "tasks, consider combining them into a single remote "
+                "function.")
+        if self.total_outbound_message_size_bytes > MESSAGE_SIZE_THRESHOLD \
+                and log_once("client_communication_overhead_warning"):
+            logger.warning(
+                "More than 10MB of messages have been created to schedule "
+                "tasks on the server. If you're running many fine-grained "
+                "tasks consider, consider combining them into a single remote "
+                "function. If you have large arguments that are frequently "
+                "reused, consider storing them remotely with ray.put or "
+                "wrapping them in an actor object.")
```

Comment:
Can you link to a reference to show code snippets or something? It's a bit hard to program from a series of text instructions, especially for new users.

Reply:
great idea